packages/trpc/server/routers/viewer/availability/schedule/duplicate.handler.ts

```diff
@@ -0,0 +1,73 @@
+import { prisma } from "@calcom/prisma";
+
+import { TRPCError } from "@trpc/server";
+
+import type { TrpcSessionUser } from "../../../../trpc";
+import type { TScheduleDuplicateSchema } from "./duplicate.schema";
+import type { Prisma } from ".prisma/client";
+
+type DuplicateScheduleOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+  input: TScheduleDuplicateSchema;
+};
+
+export const duplicateHandler = async ({ ctx, input }: DuplicateScheduleOptions) => {
+  try {
+    const { scheduleId } = input;
+    const { user } = ctx;
+    const schedule = await prisma.schedule.findUnique({
+      where: {
+        id: scheduleId,
+      },
+      select: {
+        id: true,
+        userId: true,
+        name: true,
+        availability: true,
+        timeZone: true,
+        eventType: {
+          select: {
+            id: true,
+          },
+        },
+      },
+    });
+
+    if (!schedule || schedule.userId !== user.id) {
+      throw new TRPCError({
+        code: "UNAUTHORIZED",
+      });
+    }
+
+    const { availability } = schedule;
+
+    const data: Prisma.ScheduleCreateInput = {
+      name: `${schedule.name} (Copy)`,
+      user: {
+        connect: {
+          id: user.id,
+        },
+      },
+      timeZone: schedule.timeZone ?? user.timeZone,
+      availability: {
+        createMany: {
+          data: availability.map((schedule) => ({
+            days: schedule.days,
+            startTime: schedule.startTime,
+            endTime: schedule.endTime,
```

Comment:
```suggestion
            endTime: schedule.endTime,
            date: schedule.date
```

Reply:
Thanks for the suggestion. I have added the same. I have also removed getting the `eventType` while fetching the schedule as it is not really required for duplicating, thus avoiding additional join operation.